File: 2212-removing-minimum-and-maximum-from-array/removing-minimum-and-maximum-from-array.java
class Solution {
    public int minimumDeletions(int[] nums) {
        int n = nums.length;

        int min = nums[0];
        int max = nums[0];

        for (int num : nums) {
            min = Math.min(min, num);
            max = Math.max(max, num);
        }

        int i = -1;
        int j = -1;

        for (int k = 0; k < n; k++) {
            if (nums[k] == min) {
                i = k;
            }

            if (nums[k] == max) {
                j = k;
            }
        }

        int left = Math.min(i, j);
        int right = Math.max(i, j);

        int removeLeft = right + 1;

        int removeRight = n - left;

        int removeBoth = (left + 1) + (n - right);

        return Math.min(removeLeft,
                Math.min(removeRight, removeBoth));
    }
}

File: 2212-removing-minimum-and-maximum-from-array/removing-minimum-and-maximum-from-array.py
class Solution:
    def minimumDeletions(self, nums: List[int]) -> int:
        n = len(nums)

        if n == 0:
            return 0
        if n == 1:
            return 1 

        mn = min(nums)
        mx = max(nums)

        l = nums.index(mn)
        r = nums.index(mx)
        
        idx_l = min( l,r )
        idx_r = max( l,r )
        

        del_l = idx_r+1
        del_r = n-idx_l

        del_lr = (idx_l+1)+(n-idx_r)

        return min(del_l, del_r, del_lr)





